### PR TITLE
refactor the UI of the appbar

### PR DESCRIPTION
**Pull request by devoncarew**
- refactor the UI of the appbar

Refactor the UI of the app bar to improve the aesthetics for desktop style apps. This also makes the appbar area shorter - saving space for the main content of the app.

after the change (for typical width screens):

Screen Shot 2020-03-11 at 9 51 46 AM

and, for more narrow screens:

Screen Shot 2020-03-11 at 9 52 19 AM

### CODE REVIEW

**Comment by devoncarew** [comment]
It also looks equally nice without the icons on each devtools page, just the names (like `Flutter Inspector`).

**Comment by devoncarew** [comment]
(looking into the test failures)

File: packages/devtools_app/lib/src/flutter/scaffold.dart [hunk 1]
```diff
@@ -210,15 +210,15 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
         controller: _tabController,
         isScrollable: true,
         onTap: _pushScreenToLocalPageRoute,
-        tabs: [for (var screen in widget.tabs) screen.buildTab(context)],
+        tabs: [for (var screen in widget.tabs) screen.buildTabWidget(context)],
```
Comment (jacob314) [inline packages/devtools_app/lib/src/flutter/scaffold.dart]: nit: so many things are widgets that is probably best to not prepend method names with the word widget.
For example, it is a `build` method not a `buildWidget` method.

File: packages/devtools_app/lib/src/flutter/screen.dart [hunk 2]
```diff
@@ -18,14 +18,38 @@ import 'scaffold.dart';
 /// Defines pages shown in the tabbar of the app.
 @immutable
 abstract class Screen {
-  const Screen(this.type);
+  const Screen(this.type, this.title, this.iconData, {this.tabKey});
```
Comment (jacob314) [inline packages/devtools_app/lib/src/flutter/screen.dart]: to be consistent with flutter consider renaming `iconData` to `icon` and make it a named parameter.

Comment (jacob314) [reply packages/devtools_app/lib/src/flutter/screen.dart]: also make title a named parameter

**Comment by jacob314** [comment]
Fyi @willlarche is there any material design guidance on making a material tab bar look good in a web or desktop application? This tab bar looks more natural for DevTools than the tab bar style for mobile.
For context here is what the tab bar looked like before this change.
Screen Shot 2020-03-11 at 4 35 17 PM

File: packages/devtools_app/lib/src/flutter/scaffold.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/flutter/scaffold.dart]: gotcha, will rename

File: packages/devtools_app/lib/src/flutter/screen.dart (hunk 2 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/flutter/screen.dart]: will do

**Comment by willlarche** [comment]
@clocksmith can speak to that best and perhaps have a quick conversation with a design from the Adaptive Experiences team. This may be a good opportunity to use the new Nav Rail component that is just about landed. https://github.com/flutter/flutter/pull/49574

**Comment by clocksmith** [comment]
For this particular use case, there is already some side tree navigation, so I do not know what the guidance would be for another side nav, but I will find out more

**Pull request merged**